tsdb/meta.go

```diff
@@ -515,6 +523,16 @@ func (d *DatabaseIndex) DropSeries(keys []string) {
 	atomic.AddInt64(&d.stats.NumSeries, -nDeleted)
 }
 
+// Dereference removes all references to data within b and moves them to the heap.
+func (d *DatabaseIndex) Dereference(b []byte) {
+	d.mu.Lock()
```

Comment:
`Dereference` changes the underlying series so I think we need to write lock. I'm not sure how to get around it unless we get a list of series ids and then iterate over it in chunks.